### ISSUE TITLE
strict_variables option is ignored?

### ISSUE THREAD

**Issue by kbiedrzycki** [Shopify/liquid#804]
I wanted to use strict_variables option to check if template produced by user is valid, so I've created ActiveModel validator. First step is to check if markup is valid, second is to check if variables that were used are on the allowed list.

```
template = ::Liquid::Template.parse(value, error_mode: :strict)
template.render!(allowed_variables, strict_variables: true)
```

Unfortunately, when I am passing other variables in template, no error is being thrown. 
Liquid version used `liquid (3.0.6)`.

**Comment by fw42** [comment]
can you show a more complete example that reproduces your problem? What's your template and what is your variables?

**Comment by kbiedrzycki** [comment]
@fw42 honestly? It's not matter of the template or variables, even if I try example from README:
```
template = Liquid::Template.parse("{{x}} {{y}}")
template.render!({ 'x' => 1}, { strict_variables: true })
#=> Liquid::UndefinedVariable: Liquid error: undefined variable y
```
it's not working, no error thrown at all. Also when using render without `!` errors property is not being populated.

**Comment by fw42** [comment]
This feature was added after Liquid 3.0.6 was released. The example from the README works fine for me on the current master branch.

**Comment by fw42** [comment]
It is included in v4.0.0.rc2 and v4.0.0.rc3 though, if that helps you.

**Comment by kbiedrzycki** [comment]
I've updated to rc3 and it works, thanks.
